dotCMS/src/main/java/com/dotmarketing/common/reindex/IndexResource.java

```diff
@@ -0,0 +1,381 @@
+                    @Context final HttpServletResponse response, 
+                    @DefaultValue("true") @QueryParam("switch") boolean switchMe)
+                    throws DotDataException {
+        final InitDataObject init = validateUser(request, response);
+        
+        if(!APILocator.getContentletIndexAPI().isInFullReindex()) {
+            return Response.ok(new ResponseEntityView(ESReindexationProcessStatus.getProcessIndexationMap())).build();
+        }
+        
+        if(switchMe) {
+            APILocator.getContentletIndexAPI().stopFullReindexationAndSwitchover();
+        }else {
+            APILocator.getContentletIndexAPI().stopFullReindexation();
+        }
+        return Response.ok(new ResponseEntityView(ESReindexationProcessStatus.getProcessIndexationMap())).build();
+    }
+
+    @CloseDBIfOpened
+    @POST
+    @JSONP
+    @NoCache
+    @Path("/clean")
+    public Response cleanContentType(@Context final HttpServletRequest request, @Context final HttpServletResponse response,
+                    String inode) throws DotDataException, DotSecurityException, LanguageException {
+
+        final InitDataObject init = validateUser(request, response);
+        ContentType type = APILocator.getContentTypeAPI(APILocator.systemUser()).find(inode);
+        APILocator.getContentletIndexAPI().removeContentFromIndexByStructureInode(type.id());
+        APILocator.getContentletAPI().refresh(type);
+
+        String message = LanguageUtil.get("message.cmsmaintenance.cache.indexrebuilt", init.getUser());
+
+        sendAdminMessage(message, MessageSeverity.INFO, 10000);
+
+        return Response.ok(new ResponseEntityView(message)).build();
+
+
+    }
+    @CloseDBIfOpened
+    @POST
+    @JSONP
+    @NoCache
+    @Path("/optimize")
+    public Response optimizeIndices(@Context final HttpServletRequest request, @Context final HttpServletResponse response) {
+        final InitDataObject init = validateUser(request, response);
+        ContentletIndexAPI api = APILocator.getContentletIndexAPI();
+        List<String> indices = api.listDotCMSIndices();
+        api.optimize(indices);
+        String message = Try.of(()->LanguageUtil.get(APILocator.getCompanyAPI().getDefaultCompany(),"message.cmsmaintenance.cache.indexoptimized")).get();
+
+
+        sendAdminMessage(message, MessageSeverity.INFO, 0);
+        return Response.ok().build();
+    }
+
+    @CloseDBIfOpened
+    @DELETE
+    @JSONP
+    @NoCache
+    @Path("/cache")
+    public Response flushIndiciesCache(@Context final HttpServletRequest request, @Context final HttpServletResponse response) {
+        final InitDataObject init = validateUser(request, response);
+        ContentletIndexAPI api = APILocator.getContentletIndexAPI();
+        List<String> indices = api.listDotCMSIndices();
+        
+        
+        Map<String, Integer> data = APILocator.getESIndexAPI().flushCaches(indices);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'data' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5632525)